jazzy release note: callbacks in the executor are no longer ordered. (#4459)
ros2/ros2_documentation: source/Releases/Release-Jazzy-Jalisco.rst

See https://github.com/ros2/rclcpp/pull/2142 for more details.

> Executer callbacks are no longer ordered

Author: Do we need to specifically mention SingleThreadedExecutor and StaticSingleThreadedExecutor regarding this?

Reviewer: I think what we should do here is to combine this note into the one above, which already mentions the executors.  I'll add in a suggestion for what I think we should do.

> Executer callbacks are no longer ordered

Reviewer 2: ```suggestion
Executor callbacks are no longer ordered
```

Author: Good eye! thanks!

> """"""""""""""""""""""""""""""""""""""""

Reviewer: Personally, I still think this should be combined into the point above.  It makes more sense as a single point to me.

That said, @mjcarroll , you are the expert here, so I'll defer to whatever you want to do with this.

Author: i thought that too, i will combine them into one together.

> 
> See https://github.com/ros2/rclcpp/pull/2142 for more details.
> 
> Executer callbacks are no longer ordered
> """"""""""""""""""""""""""""""""""""""""
> 
> Due to the entity collection enhancement implemented in https://github.com/ros2/rclcpp/pull/2142, callbacks in the executor are no longer ordered consistently, even within the same entity.
> 
> See the discussion https://github.com/ros2/rclcpp/issues/2532 for more details.
> 

Reviewer: ```suggestion
Due to these changes, callbacks in the executor are no longer ordered consistently, even within the same entity.

See https://github.com/ros2/rclcpp/pull/2142 and https://github.com/ros2/rclcpp/issues/2532 for more details.
```
